### PR TITLE
Rename `var_form` to `ansatz`

### PR DESCRIPTION
**Pull request by mrossinek**
### Summary

This is a follow-up to https://github.com/Qiskit/qiskit-terra/pull/6128

### Details and comments

The CI on this PR will fail, until the aforementioned PR in Terra is merged.

### CODE REVIEW

File: test/algorithms/ground_state_solvers/minimum_eigensolver_factories/test_vqe_ucc_factory.py [hunk 1]
```diff
@@ -77,10 +77,10 @@ def test_setters_getters(self):
             self.assertEqual(self._vqe_ucc_factory.include_custom, True)
 
         with self.subTest("Variational Form"):
```
Comment (pbark) [inline test/algorithms/ground_state_solvers/minimum_eigensolver_factories/test_vqe_ucc_factory.py:79]: Shall we also rename this?
```suggestion
        with self.subTest("Ansatz"):
```

File: test/algorithms/ground_state_solvers/minimum_eigensolver_factories/test_vqe_uvcc_factory.py [hunk 2]
```diff
@@ -77,10 +77,10 @@ def test_setters_getters(self):
             self.assertEqual(self._vqe_uvcc_factory.include_custom, True)
 
         with self.subTest("Variational Form"):
```
Comment (pbark) [inline test/algorithms/ground_state_solvers/minimum_eigensolver_factories/test_vqe_uvcc_factory.py:79]:
```suggestion
        with self.subTest("Ansatz"):
```

**Review by pbark — APPROVED** [review]
The rest LGTM, except the two minor things I pointed.

File: test/algorithms/ground_state_solvers/minimum_eigensolver_factories/test_vqe_ucc_factory.py (hunk 1 above)
Comment (mrossinek) [reply test/algorithms/ground_state_solvers/minimum_eigensolver_factories/test_vqe_ucc_factory.py:79]: Thanks I overlooked those! I should have done a case-insensitive search.. Will update in a second

**Review by pbark — APPROVED** [review]

**Pull request merged**